processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java

```diff
@@ -701,6 +692,24 @@ private String getName(Type type) {
             builder.append( type.getIdentification() );
             return builder.toString();
         }
+
+        private NullValueMappingStrategyPrism determineForgedNullValueMappingStrategy() {
+
+            NullValueMappingStrategyPrism nvms;
+
+            if ( method.isUpdateMethod() || TargetWriteAccessorType.GETTER == targetWriteAccessorType ) {
+                nvms = NullValueMappingStrategyPrism.RETURN_DEFAULT;
+            }
+            else {
+                if ( method.getMapperConfiguration().isMapToDefaultOnMapperAndMappingConfigLevel() ) {
```

Comment:
Couldn't you simply do
    nvms = method.getMapperConfiguration().getNullValueCheckStrategy() == NullValueMappingStrategyPrism.RETURN_DEFAULT;

That method `isMapToDefaultOnMapperAndMappingConfigLevel()`, seems a bit dubious to me.